### DISCUSSION TITLE
アプリ構成について

### DISCUSSION THREAD

**Discussion by shippokun** [ynct-d-members/todo-team#2] [Ideas]
どんなアプリにしたいのかを先にまとめてから、その後に何を使いたいのか挙げていこう。

**Comment by shippokun** [comment]
todoアプリに必要な機能と挙げていくか

**Reply by sakin97** [reply]
とりあえず基本的なCRUDができること
* todo追加
* 削除
* 編集
* 完了/未完了

**Reply by shippokun** [reply]
todoの内容はタイトルだけ？それとも詳細も書き込めるようにする？

**Reply by sakin97** [reply]
期限設定ができること

**Reply by sakin97** [reply]
> todoの内容はタイトルだけ？それとも詳細も書き込めるようにする？

詳細も書けるようにしたほうがなにかと便利そうかな

**Reply by shippokun** [reply]
タイトル
詳細
完了したか
タスクの期限


あとはいつ作ったのか？

**Reply by sakin97** [reply]
いつアップデートされたかもわかるとよさげ

**Reply by shippokun** [reply]
ユーザ認証と誰が作ったのかは要らぬか？

**Reply by sakin97** [reply]
前者はマストで、後者はチームで使う(Trelloっぽいイメージの)アプリにする方針ならほしいかな。

**Reply by shippokun** [reply]
認証したユーザのtodoのみ引っこ抜く必要があるから、紐付けは必要そう。

**Reply by sakin97** [reply]
内部での紐付けってことね、なるほど。

**Comment by shippokun** [comment]
基本的な機能をまず作ってから、そこから拡張していくか

**Comment by shippokun** [comment]
機能的なところは一旦ここでcloseしようか。
次は本題の何を使うかです🙌

**Reply by sakin97** [reply]
TypeScript!!!!!!!!

**Reply by shippokun** [reply]
PrismaORM!!!!!
https://www.prisma.io/

**Reply by shippokun** [reply]
Vue or React or Nuxt or Next or Angular どれが使いたい？

**Reply by sakin97** [reply]
思えばReactでまともなプロダクト作ったことないからReact希望

**Reply by sakin97** [reply]
SSRやるメリットあんまり思いつかないし、Nuxt/Nextまではいらないかな

**Reply by shippokun** [reply]
じゃあReactで。
cssフレームワークで希望がないならionic使ってみたい。

**Reply by shippokun** [reply]
reactやるならciで入れてみるか。とりあえず、そこまでを実際に打ち込むか。

**Reply by shippokun** [reply]
@sunakane 
今日の夜か明日あたりdiscoでしゃべりながら作業するか

**Reply by sakin97** [reply]
@shippokun すまん厳しい、日曜日はどや

**Reply by shippokun** [reply]
@sunakane ok

**Reply by sakin97** [reply]
フロントエンドルーティング
* react-router

状態管理
* redux/toolkit

**Reply by sakin97** [reply]
アプリケーションサーバ
* Fastify (https://github.com/fastify/fastify)

**Comment by shippokun** [comment]
今回PrismaORMを利用するが、apiへのアクセスはどうするのか

**Reply by shippokun** [reply]
back用のディレクトリを掘って、node.jsでサーバを立ててそいつをたたく

**Reply by shippokun** [reply]
node.jsのサーバはfastityを試験的に導入してみる
速いのは正義
もし導入が難しくてもexpressとそこまで書き方に差はないのでexpressへの移行も容易そうなので

**Comment by sakin97** [comment]
### ディレクトリ構造仮案
フロントエンド
```
src/ 
 - pages/
    - todos/    
 - store           
 - services
 - libs
 - components
```

バックエンド
ref: https://github.com/prisma/prisma-examples/tree/latest/javascript/rest-fastify
```
prisma/
 - schema
src/
```

**Reply by shippokun** [reply]
frontのpagesに関しては
```
pages/
 - todos/
  -  todo-list/
   - components/
   - containers/
   - todo-list.page.tsx
  - todo-create/
  - todo.route.tsx
```

**Comment by sakin97** [comment]
# 第一目標
reactコンポーネントのマウント時にgetリクエストを叩き，レンダリングできること

**Reply by sakin97** [reply]
マイルストーン等詳しくはProjectsにて記載

**Reply by shippokun** [reply]
まずは最低限の構成でうごくことを確認することを目標とする。
storeはまだ導入しない。

**Reply by shippokun** [reply]
コンポーネント設計は `Presentantional / Container Component` を採用する。
